ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java

```diff
@@ -1463,6 +1474,19 @@ private static String replaceTaskIdFromFilename(String filename, String oldTaskI
     return snew.toString();
   }
 
+
+  public static boolean shouldAvoidRename(FileSinkDesc conf, Configuration hConf) {
+    // we are avoiding rename/move only if following conditions are met
+    //  * execution engine is tez
+    //  * query cache is disabled
+    //  * if it is select query
+    if (conf != null && conf.getIsQuery() && conf.getFilesToFetch() != null
```

Comment:
conf will never be null

Reply:
@ashutoshc conf could be null. There is FileMerge operator which calls this code passing conf as NULL.